src/cbc_sdk/workload/vulnerability_assessment.py

```diff
@@ -0,0 +1,949 @@
+
+        Returns:
+            bool: True if refresh was successful, False if not.
+        """
+        resp = self._cb.get_object(self._build_api_request_uri())
+        resp['cve_id'] = self._model_unique_id
+        self._info = resp
+        self._last_refresh_time = time.time()
+        return True
+
+    def _build_api_request_uri(self, http_method="GET"):
+        """
+        Build the unique URL used to make requests for this object.
+
+        Args:
+            http_method (str): Not used; retained for compatibility.
+
+        Returns:
+            str: The URL used to make requests for this object.
+        """
+        return self.urlobject_single.format(self._model_unique_id)
+
+    def affected_assets(self, os_product_id):
+        """Returns a list of Vulnerability associated with device.
+
+        Args;
+            os_product_id (string) required parameter
+        Returns:
+            AffectedAssetQuery
+        """
+        if os_product_id:
+            self._os_product_id = os_product_id
+            return AffectedAssetQuery(self, self._cb)
+        else:
+            raise ApiError("The os product id is required parameter")
+
+    @classmethod
+    def _query_implementation(cls, cb, **kwargs):
+        """
+        Returns the appropriate query object for this object type.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            **kwargs (dict): Not used, retained for compatibility.
+
+        Returns:
+            AffectedAssetQuery: The query object
+        """
+        return AffectedAssetQuery(cls, cb)
+
+
+class DeviceVulnerability(NewBaseModel):
+    """DeviceVulnerability Model"""
+    urlobject = '/vulnerability/assessment/api/v1/orgs/{}/'
+    swagger_meta_file = "workload/models/vulnerability.yaml"
+    url_additional = "devices/{}/vulnerabilities/summary"
+
+    VALID_CATEGORY = ["OS", "APP"]
+
+    def __init__(self, cb, model_unique_id=None, initial_data=None):
+        """
+        Initialize DeviceVulnerability
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            model_unique_id (str): ID of the vulnerability represented.
+            initial_data (dict): Initial data used to populate the alert.
+        """
+        super(DeviceVulnerability, self).__init__(cb)
+        self._full_init = True
+
+    @classmethod
+    def _query_implementation(cls, cb, **kwargs):
+        """
+        Returns the appropriate query object for this object type.
+
+        Args:
+            cb (BaseAPI): Reference to API object used to communicate with the server.
+            **kwargs (dict): Not used, retained for compatibility.
+
+        Returns:
+            DeviceVulnerabilityQuery: The query object
+        """
+        return DeviceVulnerabilityQuery(cls, cb)
+
+    @classmethod
+    def get_vulnerability_summary_per_device(cls, cb, device, category=None, vcenter_id=None):
```

Comment:
This should be device_id instead of a device object. The doc_string is also missing some of the parameters.

Reply:
Agreeing with Alex: use device_id (int) here instead of device (object).